src/lib/decentraland/crdt-internal/last-write-win-element-set.ts

```diff
@@ -154,38 +143,34 @@ export function createGetCrdtMessagesForLww<T>(
   serde: SerDe<T>,
   data: Map<Entity, T>
 ) {
-  return function* () {
+  return function (outBuffer: ByteBuffer) {
     for (const entity of dirtyIterator) {
       const newTimestamp = incrementTimestamp(entity, timestamps)
       if (data.has(entity)) {
         const writeBuffer = new ReadWriteByteBuffer()
         serde.serialize(data.get(entity)!, writeBuffer)
 
-        const msg: PutComponentMessageBody = {
-          type: CrdtMessageType.PUT_COMPONENT,
+        PutComponentOperation.write(
+          entity,
           componentId,
-          entityId: entity,
-          data: writeBuffer.toBinary(),
-          timestamp: newTimestamp
-        }
-
-        yield msg
+          newTimestamp,
+          writeBuffer.toBinary(),
+          outBuffer
+        )
```

Comment:
A thought: would this be easier to follow / reason about / play with / test if there was an intermediate object?

```ts
// Option 1: with orderered parameters (warning, two consecutive ones are `number`)
new PutComponentOperation(entity, componentId, timestamp, state).write(buffer)

// Option 2: with named fields
new PutComponentOperation({ entity, componentId, timestamp, state }).write(buffer)
```

We could eliminate the paramater mix, and give users an intermediate object that can be inspected, logged, etc. Could be useful for people experimenting with this code.